File: tutorials/csharp-tutorial/src/CSharpTutorialDemo/CSharpTutorialDemo/Inheritance.cs
using System;

namespace CSharpTutorialDemo
{
    public class Inheritance
    {
        class Animal
        {
            public string Name { get; set; }

            public Animal(string name)
            {
                Name = name;
            }
            public void PrintName()
            {
                Console.WriteLine(Name);
            }
        }

        class Cat : Animal
        {
            public Cat(string name) : base (name)
            {

            }

            public void Meow()
            {
                Console.WriteLine("Meow!");
            }
        }

        class Dog : Animal
        {
            public Dog(string name) : base(name)
            {

            }

            public void Bark()
            {
                Console.WriteLine("Bark!");
            }
        }

        public static void Example1()
        {
            Cat cat = new Cat("Stanley");
            Dog dog = new Dog("Jackie");

            cat.PrintName();
            cat.Meow();

            dog.PrintName();
            dog.Bark();
        }
    }
}

File: tutorials/vb-net/src/VBNetTutorialDemo/VBNetTutorialDemo/Inheritance.vb
Module Inheritance
    Class Animal
        Public Property Name As String

        Public Sub New(ByVal name As String)
            name = name
        End Sub

        Public Sub PrintName()
            Console.WriteLine(Name)
        End Sub
    End Class

    Class Cat
        Inherits Animal

        Public Sub New(ByVal name As String)
            MyBase.New(name)
        End Sub

        Public Sub Meow()
            Console.WriteLine("Meow!")
        End Sub
    End Class

    Class Dog
        Inherits Animal

        Public Sub New(ByVal name As String)
            MyBase.New(name)
        End Sub

        Public Sub Bark()
            Console.WriteLine("Bark!")
        End Sub
    End Class

    Public Sub Example1()
        Dim cat As Cat = New Cat("Stanley")
        Dim dog As Dog = New Dog("Jackie")
        cat.PrintName()
        cat.Meow()
        dog.PrintName()
        dog.Bark()
    End Sub
End Module
